spec/features/admin/invoice/show_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe 'admin/invoices/invoice.id' do
+  before :each do
+    @customer_1 = Customer.create!(first_name: 'Eli', last_name: 'Fuchsman')
+
+    @merchant = Merchant.create!(name: 'Test')
+
+    @item_1 = Item.create!(name: 'item1', description: 'desc1', unit_price: 10, merchant_id: @merchant.id)
+    @item_2 = Item.create!(name: 'item2', description: 'desc2', unit_price: 12, merchant_id: @merchant.id)
+
+    @invoice_1 = Invoice.create!(customer_id: @customer_1.id, status: 1)
+
+    @ii_1 = InvoiceItem.create!(invoice_id: @invoice_1.id, item_id: @item_1.id, quantity: 9, unit_price: 10, status: 1)
+    @ii_1 = InvoiceItem.create!(invoice_id: @invoice_1.id, item_id: @item_2.id, quantity: 11, unit_price: 12, status: 1)
+
+    @transaction_1 = Transaction.create!(credit_card_number: '1', result: 0, invoice_id: @invoice_1.id)
+    @transaction_2 = Transaction.create!(credit_card_number: '1', result: 0, invoice_id: @invoice_1.id)
+  end
+  describe 'invoice show, has invoice info, invoice items info, and total revenue' do
+    it 'shows related information to a specific invoice' do
+      visit "/admin/invoices/#{@invoice_1.id}"
+      expect(page).to have_content("Id: #{@invoice_1.id}")
+      expect(page).to have_content('Status: completed')
+      expect(page).to have_content('Created at: Saturday, November 5, 2022')
```

Comment:
I think this test will fail when it is not November 5th. You will probably have to manually add the created_at date or use mock/stub here.

Reply:
Updated with Darby and I on Zoom.